### PR TITLE
As/zendao optimization

### PR DESCRIPTION
**Pull request by alsala**
## Description
Optimize time and gas cost in Zendao Native Smart Contract when checking mc address association

## Jira Ticket
https://horizenlabs.atlassian.net/browse/SDK-1152

## Changes
Avoid looping on the full list of associations in order to check if a mc address is already associated to a sc address.
Added py test for checking that adding and removing associations requires a constant gas amount.

### CODE REVIEW

File: sdk/src/test/scala/io/horizen/account/state/McAddrOwnershipMsgProcessorTest.scala [hunk 1]
```diff
@@ -304,10 +304,10 @@ class McAddrOwnershipMsgProcessorTest
...
         scAddressObj1
       )
-      val expectedOwnershipId = Keccak256.hash(Bytes.concat(scAddressObj1.toBytes, mcAddrStr1.getBytes(StandardCharsets.UTF_8)))
+      val expectedOwnershipId = Keccak256.hash(mcAddrStr1.getBytes(StandardCharsets.UTF_8))
 
       // positive case, verify we can add the data to view
-      val returnData = assertGas(181037, msg, view, messageProcessor, defaultBlockContext)
+      val returnData = assertGas(180937, msg, view, messageProcessor, defaultBlockContext)
       assertNotNull(returnData)
       println("This is the returned value: " + BytesUtils.toHexString(returnData))
```
Comment (paologalligit) [inline sdk/src/test/scala/io/horizen/account/state/McAddrOwnershipMsgProcessorTest.scala:312]: Can we remove this println statement?

File: sdk/src/test/scala/io/horizen/account/state/McAddrOwnershipMsgProcessorTest.scala [hunk 2]
```diff
@@ -342,7 +342,7 @@ class McAddrOwnershipMsgProcessorTest
       val txHash3 = Keccak256.hash("third tx")
       view.setupTxContext(txHash3, 10)
 
-      val returnData2 = assertGas(198737, msg2, view, messageProcessor, defaultBlockContext)
+      val returnData2 = assertGas(189837, msg2, view, messageProcessor, defaultBlockContext)
       assertNotNull(returnData2)
       println("This is the returned value: " + BytesUtils.toHexString(returnData2))
```
Comment (paologalligit) [inline sdk/src/test/scala/io/horizen/account/state/McAddrOwnershipMsgProcessorTest.scala:347]: Can we remove this println statement?

File: qa/sc_evm_mc_addr_ownership_perf_test.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        current_best_epoch = sc_node.block_forgingInfo()["result"]["bestEpochNumber"]
+
+        for i in range(0, ZENDAO_FORK_EPOCH - current_best_epoch):
+            generate_next_block(sc_node, "first node", force_switch_to_next_epoch=True)
+            self.sc_sync_all()
+
+        # try adding many mc addresses and forge a block after that
+
+        # this can take long time
+        # num_of_association = 10000
```
Comment (paologalligit) [inline qa/sc_evm_mc_addr_ownership_perf_test.py:90]: Is this comment still useful?

**Review by paologalligit — APPROVED** [review]
Left some minor comments but overall looks good

**Review by paolocappelletti — APPROVED** [review]
looks good!

**Review by MarcoOl94 — APPROVED** [review]

**Pull request merged**